docs/src/proposals/optimistic_confirmation.md

```diff
@@ -0,0 +1,237 @@
+# Optimistic Confirmation
+
+## Primitives
+
+`vote(X, S)` - Votes will be augmented with a "reference" slot, `X` 
+which is the **last** ancestor at which this validator voted on this fork
+with a proof of switching. Thus as long as a validator keeps voting on the 
+same fork, `X` will remain unchanged.All votes will then be of the form
+`vote(X, S)`, where `S` is the sorted list of slots `(s, s.lockout)`
```

Comment:
Is S the entire Tower stack?

Reply:
@aeyakovenko, I think that will depend on how we define the contents of the proof of switching, specifically how we intend prove the amount of "lockout" a node has committed to a specific slot `s`.

Right now the section about the contents of the `proof of switching` here: https://github.com/solana-labs/solana/pull/9157/files#diff-af0aef1677e9797c44811a991da7fa2eR54-R64

 is pretty hand-wavy in that it assumes there's some to show in a `proof of switching` that validator `X` is locked out on slot `s` at slot `s'`, i.e. show that `s + s.lockout > s'`. 

This means either introducing some sort of new `lockout` field to our current votes (how do you verify a node is not lying about their lockout), or the proof of switching will need to include a series of votes that demonstrate lockout was built up.